src/disk.c

```diff
@@ -981,12 +1006,8 @@ static int disk_read(void) {
       metric_family_metric_append(&fam_disk_io_weighted_time, m);
 
       long interval = CDTIME_T_TO_MS(plugin_get_interval());
-      if (interval == 0) {
-        DEBUG("disk plugin: got zero plugin interval");
-      }
-
-      m.value.gauge = ((diff_io_time / (double)interval) * 100.0);
-      metric_family_metric_append(&fam_disk_utilization, m);
+      m.value.gauge = ((gauge_t)diff_io_time) / ((gauge_t)interval);
```

Comment:
div-by-zero if interval is zero?

Reply:
I think we can assume that the interval is non-zero. Otherwise I think a lot will break in the daemon. We divide by the interval somewhat frequently to calculate rates.